apiserver/services/serviceregistry.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package services
+
+import (
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain"
+	controllerconfigservice "github.com/juju/juju/domain/controllerconfig/service"
+	controllerconfigstate "github.com/juju/juju/domain/controllerconfig/state"
+	controllernodeservice "github.com/juju/juju/domain/controllernode/service"
+	controllernodestate "github.com/juju/juju/domain/controllernode/state"
+	externalcontrollerservice "github.com/juju/juju/domain/externalcontroller/service"
+	externalcontrollerstate "github.com/juju/juju/domain/externalcontroller/state"
+	modelmanagerservice "github.com/juju/juju/domain/modelmanager/service"
+	modelmanagerstate "github.com/juju/juju/domain/modelmanager/state"
+)
+
+// Logger defines the logging interface used by the services.
+type Logger interface {
+	Debugf(string, ...interface{})
+	Child(string) Logger
+}
+
+// Registry provides access to the services required by the apiserver.
+type Registry struct {
+	controllerDB func() (changestream.WatchableDB, error)
+	deleterDB    database.DBDeleter
```

Comment:
So here. How is it set?

Reply:
OK, you like the idea, I'll finish it off now.